Using `state: details` seems to cause this PR to fail.

```
  - name: "Detail environment (DETAIL ENVIRONMENT)"
    elasticbeanstalk_env:
      region: "{{ region }}"
      app_name: "{{ app_name }}"
      env_name: "{{ environment_name }}"
      state: "details"
```

```
TASK [Detail environment (DETAIL ENVIRONMENT)] *********************************
task path: /vagrant/ansible-elastic-beanstalk/tests/elasticbeanstalk_env-tests.yml:205
Using module file /vagrant/ansible-elastic-beanstalk/library/elasticbeanstalk_env.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: vagrant
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1484806588.66-114928600384571 `" && echo ansible-tmp-1484806588.66-114928600384571="` echo $HOME/.ansible/tmp/ansible-tmp-1484806588.66-114928600384571 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmp7L36AD TO /home/vagrant/.ansible/tmp/ansible-tmp-1484806588.66-114928600384571/elasticbeanstalk_env.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/vagrant/.ansible/tmp/ansible-tmp-1484806588.66-114928600384571/ /home/vagrant/.ansible/tmp/ansible-tmp-1484806588.66-114928600384571/elasticbeanstalk_env.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1484806588.66-114928600384571/elasticbeanstalk_env.py; rm -rf "/home/vagrant/.ansible/tmp/ansible-tmp-1484806588.66-114928600384571/" > /dev/null 2>&1 && sleep 0'
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_eyqQI4/ansible_module_elasticbeanstalk_env.py", line 410, in <module>
    main()
  File "/tmp/ansible_eyqQI4/ansible_module_elasticbeanstalk_env.py", line 340, in main
    env = describe_env_config_settings(ebs, app_name, env_name, [])
TypeError: describe_env_config_settings() takes exactly 3 arguments (4 given)

fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "invocation": {
        "module_name": "elasticbeanstalk_env"
    },
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_eyqQI4/ansible_module_elasticbeanstalk_env.py\", line 410, in <module>\n    main()\n  File \"/tmp/ansible_eyqQI4/ansible_module_elasticbeanstalk_env.py\", line 340, in main\n    env = describe_env_config_settings(ebs, app_name, env_name, [])\nTypeError: describe_env_config_settings() takes exactly 3 arguments (4 given)\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE"
}
```

Actual terminate action appears to be fixed however.

It also appears that it'll display the terminated environment if you try to list it after termination, I think you'd probably just want to say not found.

```
TASK [List environment (NOT FOUND)] ********************************************
task path: /vagrant/ansible-elastic-beanstalk/tests/elasticbeanstalk_env-tests.yml:232
Using module file /vagrant/ansible-elastic-beanstalk/library/elasticbeanstalk_env.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: vagrant
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1484809531.35-121401639497293 `" && echo ansible-tmp-1484809531.35-121401639497293="` echo $HOME/.ansible/tmp/ansible-tmp-1484809531.35-121401639497293 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpLumDPX TO /home/vagrant/.ansible/tmp/ansible-tmp-1484809531.35-121401639497293/elasticbeanstalk_env.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/vagrant/.ansible/tmp/ansible-tmp-1484809531.35-121401639497293/ /home/vagrant/.ansible/tmp/ansible-tmp-1484809531.35-121401639497293/elasticbeanstalk_env.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1484809531.35-121401639497293/elasticbeanstalk_env.py; rm -rf "/home/vagrant/.ansible/tmp/ansible-tmp-1484809531.35-121401639497293/" > /dev/null 2>&1 && sleep 0'
ok: [localhost] => {
    "changed": false,
    "env": {
        "AbortableOperationInProgress": false,
        "ApplicationName": "elasticbeanstalk_env-testing",
        "CNAME": "elasticbeanstalk-env-testing-79.us-west-2.elasticbeanstalk.com",
        "DateCreated": "2017-01-19T06:54:30.120000+00:00",
        "DateUpdated": "2017-01-19T07:05:20.452000+00:00",
        "EndpointURL": "internal-awseb-e-s-AWSEBLoa-1QBX4UO6H17JD-1044323694.us-west-2.elb.amazonaws.com",
        "EnvironmentId": "e-saby4feqf2",
        "EnvironmentLinks": [],
        "EnvironmentName": "elasticbeanstalk-env-testing-79",
        "Health": "Grey",
        "SolutionStackName": "64bit Amazon Linux 2016.09 v2.3.1 running Tomcat 8 Java 8",
        "Status": "Terminated",
        "Tier": {
            "Name": "WebServer",
            "Type": "Standard",
            "Version": "1.0"
        },
        "VersionLabel": "Testing"
    },
    "invocation": {
        "module_args": {
            "app_name": "elasticbeanstalk_env-testing",
            "aws_access_key": null,
            "aws_secret_key": null,
            "cname_prefix": null,
            "description": null,
            "ec2_url": null,
            "env_name": "elasticbeanstalk-env-testing-79",
            "option_settings": [],
            "options_to_remove": [],
            "profile": null,
            "region": "us-west-2",
            "security_token": null,
            "solution_stack_name": null,
            "state": "list",
            "tags": {},
            "template_name": null,
            "tier_name": "WebServer",
            "validate_certs": true,
            "version_label": null,
            "wait_timeout": 900
        },
        "module_name": "elasticbeanstalk_env"
    }
}
```

--- a/library/elasticbeanstalk_env.py
+++ b/library/elasticbeanstalk_env.py
@@ -337,7 +337,7 @@ def main():
 
     if state == 'details':
         try:
-            env = describe_env_config_settings(ebs, app_name, env_name, [])
+            env = describe_env_config_settings(ebs, app_name, env_name)
             result = dict(changed=False, env=env)
         except ClientError, e:
             module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))

